checkov/sast/checks/checks_parser.py

```diff
@@ -0,0 +1,137 @@
+            logging.error(f'cant parse the following policy: {raw_check}')
+        try:
+            semgrep_rule = self.parse_rule_metadata(raw_check, semgrep_rule)
+            semgrep_rule.update(self.parse_definition(raw_check['definition']))
+        except Exception as e:
+            raise e
+
+        return semgrep_rule
+
+    def raw_check_is_valid(self, raw_check: Dict[str, Any]) -> bool:
+        metadata = raw_check.get('metadata')
+        if not metadata:
+            raise AttributeError('BQL policy is missing the metadata field')
+        if not metadata.get('id'):
+            raise AttributeError('BQL policy metadata is missing an id value')
+        if not metadata.get('severity'):
+            raise AttributeError('BQL policy metadata is missing a severity')
+        if not raw_check.get('scope', {}).get('languages'):
+            raise AttributeError('BQL policy metadata is missing languages')
+
+        else:
+            return True
+
+    def parse_rule_metadata(self, bql_policy: Dict[str, Any], semgrep_rule: Dict[str, Any]) \
+            -> Dict[str, Any]:
+        metadata = bql_policy['metadata']
+        semgrep_rule[SemgrepAttribute.ID.value] = metadata['id']
+        semgrep_rule[SemgrepAttribute.MESSAGE.value] = metadata.get('guidelines', '')
+        semgrep_rule[SemgrepAttribute.SEVERITY.value] = CHECKOV_SEVERITY_TO_SEMGREP_SEVERITY[metadata['severity']]
+        languages = bql_policy['scope']['languages']
+        semgrep_rule[SemgrepAttribute.LANGUAGES.value] = languages
+        semgrep_rule['metadata'] = {'name': metadata['name']}
+
+        # add optional cwe and owasp metadata
+        cwe = metadata.get('cwe')
+        if cwe:
+            semgrep_rule['metadata'][SemgrepAttribute.CWE.value] = cwe
+        owasp = metadata.get('owasp')
+        if owasp:
+            semgrep_rule['metadata'][SemgrepAttribute.OWASP.value] = owasp
+
+        return semgrep_rule
+
+    def parse_definition(self, definition: Dict[str, Any]) -> Dict[str, Any]:
+        definitions = force_list(definition)
+        conf = {}
+        if len(definitions) > 1:
+            conf[SemgrepAttribute.PATTERNS.value] = self.get_definitions_list_items(definitions)
+        else:  # since definitions has only one value, we check only definitions[0]
+            definition = definitions[0]
+            if not isinstance(definition, dict):
+                raise TypeError(f'bad definition type, got {type(definition)} instead of dict')
+            if definition.get(BqlConditionType.OR.value):
+                conf[SemgrepAttribute.PATTERN_EITHER.value] = self.get_definitions_list_items(
+                    definition[BqlConditionType.OR.value])
+            elif definition.get(BqlConditionType.AND.value):
+                conf[SemgrepAttribute.PATTERNS.value] = self.get_definitions_list_items(
+                    definition[BqlConditionType.AND.value])
+            else:
+                # handle a single definition
```

Comment:
perhaps separate this block of code to a new function called `_parse_single_definition`? It is already pretty long.

Overall you can really simplify this function and make it more readable if you separate each big block of "elif" and call it by the condition.
For example on line 93 separate the block to `_parse_condition_with_variable`

Reply:
Great idea